Sources/BraintreeThreeDSecure/BTThreeDSecureClient.swift

```diff
@@ -260,6 +259,8 @@ import BraintreeCore
     
     private func process(lookupResult: BTThreeDSecureResult, configuration: BTConfiguration) {
         if lookupResult.lookup?.requiresUserAuthentication == false || lookupResult.lookup == nil {
```

Comment:
I didn't understand the `lookupResult.lookup == nil` condition. Is this considered the same as acsURL being nil? This `lookupResult.lookup == nil` is not an error in this flow?

Reply:
This is just saying if the `lookupResult.lookup == nil` we didn't get back a 3DS lookup object so can continue as if we aren't requiring 3D Secure.

If we look at the 2 call sites for this method:
```swift
private func start(request: BTThreeDSecureRequest, configuration: BTConfiguration) {
    ...
    guard let lookupResult, error == nil else { ... }
    ...
    self.process(lookupResult: lookupResult, configuration: configuration)
}
```
Here we would always have a `lookupResult.lookup` since we are doing a guard on that property to check that it's not nil.

The second call site for this method is here:
```swift
public func initializeChallenge(
        lookupResponse: String,
        request: BTThreeDSecureRequest,
        completion: @escaping (BTThreeDSecureResult?, Error?) -> Void
    ) {
    ...
    let jsonResponse = BTJSON(data: dataResponse)
    let lookupResult = BTThreeDSecureResult(json: jsonResponse)
    ...
    self.process(lookupResult: lookupResult, configuration: configuration)
}
```
In this case (server side 3DS calls for large merchants) there could be an instance where when creating the `BTThreeDSecureResult` from the JSON data passed from the merchant it wouldn't contain the `lookupResult.lookup` property and thus hit the check `lookupResult.lookup == nil` in this method.

Reply:
What I was wondering was is it an expected behavior that lookup object is nil? Would that be something intentional? If not, would it be an error? I see that we set requiresAuthentication field to the acsURL being nil. Above, we are not doing guard against lookupResult.lookup but lookupResult itself.

Reply:
> Above, we are not doing guard against lookupResult.lookup but lookupResult itself.

Ah, yep you are right - see still getting my brain back. 😄 

> What I was wondering was is it an expected behavior that lookup object is nil? Would that be something intentional? If not, would it be an error? I see that we set requiresAuthentication field to the acsURL being nil.

It can be `nil` without causing issues. It would just mean that we wouldn't attempt a 3D Secure challenge. But there are any number of reasons that we may not get this object back and as such treat it the same way we treat not requiring 3D Secure. It would not be an error, it just means we don't attempt or launch a 3D Secure challenge.

Reply:
I'm just glad that you are back, Jax. We missed you. I am a bit confused. In the OR condition in `process` function, that block returns success. Shouldn't success here be intentional, such as lookup with no acsURL?